pkg/resources/preprocessor/workloads.go

```diff
@@ -70,31 +72,39 @@ processors:
     log_statements:
     - context: log
       statements:
-      - merge_maps(attributes, body, "upsert") where attributes["filename"] == nil
       - set(attributes["COMM"], attributes["_COMM"])
       - delete_matching_keys(attributes, "^_.*")
     - context: resource
       statements:
       - set(attributes["log_type"], "controlplane") where attributes["k8s.pod.labels.tier"] == "control-plane"
       - set(attributes["kubernetes_component"], attributes["k8s.pod.labels.component"]) where attributes["k8s.pod.labels.tier"] == "control-plane"
+
 exporters:
   opensearch:
-    endpoints: [ "{{ .Endpoint }}" ]
-    index: {{ .WriteIndex }}
+    http:
+      endpoint: {{ .Endpoint }}
+      tls:
+        ca_file: /etc/otel/chain.crt
+        cert_file: /etc/otel/certs/tls.crt
+        key_file: /etc/otel/certs/tls.key
+    logs_index: {{ .WriteIndex }}
+    dataset: kubernetes
+    namespace: opni
     mapping:
       mode: flatten_attributes
       timestamp_field: time
       unix_timestamp: true
-    tls:
-      ca_file: /etc/otel/chain.crt
-      cert_file: /etc/otel/certs/tls.crt
-      key_file: /etc/otel/certs/tls.key
+      dedup: true
+      dedot: false
 service:
   pipelines:
     logs:
       receivers: ["otlp"]
       processors: ["resource", "attributes", "transform"]
       exporters: ["opensearch"]
+    traces:
+      receivers: ["otlp"]
```

Comment:
I ended up not adding any processors for the trace pipelines. Will do them in https://github.com/rancher/opni/issues/1748.